chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/UserDeveloperMapDAOImpl.java

```diff
@@ -0,0 +1,183 @@
+package gov.healthit.chpl.dao.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.stereotype.Repository;
+
+import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.dao.UserDeveloperMapDAO;
+import gov.healthit.chpl.dto.UserDeveloperMapDTO;
+import gov.healthit.chpl.entity.developer.DeveloperEntity;
+import gov.healthit.chpl.entity.UserDeveloperMapEntity;
+import gov.healthit.chpl.entity.auth.UserEntity;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+
+@Repository(value = "userDeveloperMapDAO")
+public class UserDeveloperMapDAOImpl extends BaseDAOImpl implements UserDeveloperMapDAO {
+
+    @Override
+    public UserDeveloperMapDTO create(UserDeveloperMapDTO dto) throws EntityRetrievalException {
+        UserDeveloperMapEntity entity = new UserDeveloperMapEntity();
+        entity = create(getUserDeveloperMapEntity(dto));
+        return new UserDeveloperMapDTO(entity);
+    }
+
+    @Override
+    public void delete(UserDeveloperMapDTO dto) throws EntityRetrievalException {
+        UserDeveloperMapEntity entity = getEntityById(dto.getId());
+        entity.setDeleted(true);
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        update(entity);
+    }
+
+    @Override
+    public List<UserDeveloperMapDTO> getByUserId(final Long userId) {
+        Query query = entityManager.createQuery(
+                "from UserDeveloperMapEntity udm "
+                        + "join fetch udm.developer developer "
+                        + "join fetch udm.user u "
+                        + "join fetch u.permission perm "
+                        + "join fetch u.contact contact "
+                        + "where (udm.deleted != true) AND (u.id = :userId) ",
+                UserDeveloperMapEntity.class);
+        query.setParameter("userId", userId);
+        List<UserDeveloperMapEntity> result = query.getResultList();
+
+        List<UserDeveloperMapDTO> dtos = new ArrayList<UserDeveloperMapDTO>();
+        if (result != null) {
+            for (UserDeveloperMapEntity entity : result) {
+                dtos.add(new UserDeveloperMapDTO(entity));
+            }
+        }
+        return dtos;
+    }
+
+    @Override
+    public List<UserDeveloperMapDTO> getByDeveloperId(final Long developerId) {
+        Query query = entityManager.createQuery(
+                "from UserDeveloperMapEntity udm "
+                        + "join fetch udm.developer developer "
+                        + "join fetch udm.user u "
+                        + "join fetch u.permission perm "
+                        + "join fetch u.contact contact "
+                        + "where (udm.deleted != true) "
+                        + "AND (cb.id = :developerId) ",
```

Comment:
It says "cb.id" but I don't think "cb" is anything in this query. So I think either change the HQL or if the code is not being called maybe we don't need this method?